app_llm.py

```diff
@@ -356,56 +375,82 @@ def set_first_row_as_header(df):
-                st.session_state.table_names = table_names
-            if 'table_dfs' not in st.session_state:
-                st.session_state.table_dfs = table_dfs
-
-            # Displaying the editable information
-            for i, (table_name, df) in enumerate(zip(st.session_state.table_names, st.session_state.table_dfs)):
-                st.write(f"{table_name}")
-                col1, col2 = st.columns([4, 1])
-
-                with col1:
-                    # Display tables as editable fields
-                    table_dfs[i] = st.data_editor(df, num_rows="dynamic", key=f"editor_{i}", use_container_width=True)
-
-                with col2:
-                    # Add column functionality
-                    # new_col_name = st.text_input(f"New column name", key=f"new_col_{i}")
-                    if st.button(f"Add Column", key=f"add_col_{i}"):
-                        table_dfs[i][int(table_dfs[i].columns[-1]) + 1] = None
-
-                    # Delete column functionality
-                    if not st.session_state.table_dfs[i].empty:
-                        col_to_delete = st.selectbox(f"Column to delete", st.session_state.table_dfs[i].columns,
-                                                    key=f"del_col_{i}")
-                        if st.button(f"Delete Column", key=f"delete_col_{i}"):
-                            table_dfs[i] = table_dfs[i].drop(columns=[col_to_delete])
-            
-            # This can normalize table headers to match DHIS2 using Levenstein distance or semantic search
-            # TODO: Currently there's only a small set of hard coded fields, which might look weird to the user, so it's left of for the demo
-            #if st.button(f"Correct field names", key=f"correct_names"):
-            #     table_dfs = correct_field_names(table_dfs)
-                
-            # Rerun the code to display any edits made by user
-            for idx, table in enumerate(table_dfs):
-                if not table_dfs[idx].equals(st.session_state.table_dfs[idx]):
-                    st.session_state.table_dfs = table_dfs
-                    st.rerun()
-            
-            if 'data_payload' not in st.session_state:
-                st.session_state.data_payload = None
+        if 'table_names' not in st.session_state:
+            st.session_state.table_names = table_names
+        if 'table_dfs' not in st.session_state:
+            st.session_state.table_dfs = table_dfs
+        if 'page_nums' not in st.session_state:
+            st.session_state.page_nums = page_nums
+
+        # Displaying the editable information
+        
+        page_options = {num for num in st.session_state.page_nums}
+        
+        page_selected = st.selectbox("Page Number", page_options)
+        
+        # Displaying images so the user can see them
+        with st.expander("Show Image"):
+            sheet = tally_sheet[int(str(page_selected).replace(" ✓", "")) - 1]
```

Comment:
This is a really minor thing that I just noticed, but since now the user can upload multiple images, I'm going to make `tally_sheet` plural as `tally_sheets` just to make it clearer that its a list. I don't think it was introduced in this PR, but this was the first time it made it harder for me to understand the code, so I'm going to change it now.